lib/hatchet.rb

```diff
@@ -26,6 +26,12 @@ module Hatchet
   Runner  = Hatchet::GitApp
 
   def self.git_branch
+    # https://circleci.com/docs/variables
+    return ENV['CIRCLE_BRANCH'] if ENV['CIRCLE_BRANCH']
+    # https://docs.github.com/en/actions/learn-github-actions/environment-variables
+    return ENV['GITHUB_REF_NAME'] if ENV['GITHUB_REF_NAME']
```

Comment:
So I tried using `GITHUB_REF_NAME` directly in my Python GHA PR, however for PRs it resulted in buildpack URLs like:
`https://github.com/heroku/heroku-buildpack-python.git#1371/merge`

It seems we have to use:
- `GITHUB_HEAD_REF` for PRs
- and then `GITHUB_REF_NAME` otherwise (which covers the branches and tags cases)

We can't use `GITHUB_HEAD_REF` for both, since it's only set for PRs.

Reply:
You can see the available env vars in these runs were I printed the entire env:
- branch job: https://github.com/heroku/heroku-buildpack-python/actions/runs/3159749152/jobs/5143363024
- PR job: https://github.com/heroku/heroku-buildpack-python/actions/runs/3159749360/jobs/5143363670

Reply:
This is what I used as a workaround in the meantime:

Reply:
This is what I used as a workaround in the meantime:
https://github.com/heroku/heroku-buildpack-python/blob/714d84e81dea1dbd58ba75cd34183a7082d501f7/.github/workflows/ci.yml#L40-L42

Reply:
> Ready to merge?

I guess not until this is ironed out maybe something like this?

```
ENV["GITHUB_REF_NAME"].delete_suffix("/merge")
```

Reply:
Also worth noting that this should fall back to using `git` to detect the current branch. It's just some CI providers don't give you the full history with branch name etc. I'm not sure about github actions

Reply:
I added @edmorley's suggestion above. We now prefer `GITHUB_HEAD_REF` if it's present, since that's the correct branch name for PRs. If that value is blank (which it seems to be on branch builds), we'll fallback to `GITHUB_REF_NAME`.